realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -625,6 +625,11 @@ private static void initializeRealm(Realm realm) {
         if (clazz == null || inputStream == null) {
             return null;
         }
+        Table table = getTable(clazz);
+        if (table.hasPrimaryKey()) {
+            throw new UnsupportedOperationException("Cannot create RealmObjects with primary keys using Streams. " +
+                    "Use createOrUpdateObjectFromJson() instead.");
```

Comment:
I am not sure how many people are using our JSONStream interface, but this would create a really bad user experience :( What about we use standalone object to set all values first?